packages/@aws-cdk/aws-ec2/lib/bastion-host.ts

```diff
@@ -176,13 +177,23 @@ export class BastionHostLinux extends Resource implements IInstance {
     this.instancePrivateDnsName = this.instance.instancePrivateDnsName;
     this.instancePublicIp = this.instance.instancePublicIp;
     this.instancePublicDnsName = this.instance.instancePublicDnsName;
+    this.applyUserData();
 
     new CfnOutput(this, 'BastionHostId', {
       description: 'Instance ID of the bastion host. Use this to connect via SSM Session Manager',
       value: this.instanceId,
     });
   }
 
+  /**
+   * Install the architecture-appropriate SSM Agent package
+   */
+  private applyUserData(): void {
+    const instanceType = new InstanceType(this.instance.instance.instanceType!);
```

Comment:
Seems safer and easier to have `instanceType` passed in here from the constructor.

Reply:
I confirmed that the SSM Agent comes preinstalled on AL2 for both x86-64 and aarch64, so we have an opportunity to simplify. I have removed the userdata logic.